### ISSUE TITLE
Heat flux computation issue

### ISSUE THREAD

**Issue by aeroamit** [su2code/SU2#1063]
**Describe the bug**
*I tried to compute heat flux over a hemisphere at Mach 5 for calorically perfect gas (axisymmetric simulation), results are incorrect in the stagnation region*

**To Reproduce**
Files attached along with pdf containing my results/details
[hf_issue.pdf](https://github.com/su2code/SU2/files/5031496/hf_issue.pdf)
[hemi.zip](https://github.com/su2code/SU2/files/5031499/hemi.zip)

**Desktop (please complete the following information):**
 - OS: [e.g., Linux (Ubuntu 20.04)]
 - C++ compiler and version: [latest]
 - MPI implementation and version: [latest]
 - SU2 Version: [v7.0.6]

**Comment by WallyMaier** [comment]
I have noticed issues here as well.  It could be due to the axisymmetric source terms not including viscous effects.  I was having issues when running Euler at Mach>15 using the axisymmetric source terms

**Comment by aeroamit** [comment]
Hi Wally,

In that case, I have to find one suitable 2D case. Can you / anyone suggest an appropriate case for heat flux computation at higher Mach numbers.

Regards

**Comment by WallyMaier** [comment]
I do not have a specific test case in mind unfortunately, most cases I have looked at have been Euler/multispecies.  If you dont mind running turbulence, you can try looking into some flat plate simulations. Sorry, Im not more help here

**Comment by aeroamit** [comment]
I will figure out something and try 2D case

**Comment by aeroamit** [comment]
Finally I converted the above axisymmetric case (sphere) to to symmetry case (cylinder) and compared the results between SU2 and alternate code (CFD++). Conditions are M=5, Tinf=Twall=212K. 
[Comparison.pdf](https://github.com/su2code/SU2/files/5250316/Comparison.pdf)

Here Cp compares well. Heat flux comparison is  reasonable (minor differences, which should not be a worry).

As Wally mentioned, there are issues with axisymmtric implementation. Let us fix it. I feel it is very important, as many folks must be making use of axisymmetric formulation for various problems.

**Comment by WallyMaier** [comment]
Hi Amit, to further the discussion, I am running a sharp cone axisymmetric (or at least testing it).  It makes me wonder if the issue only appears where the axis of rotation is perpendicular with a wall.  Also, the axisymmetric boundaries havent been formulated for NS sims yet.

**Comment by aeroamit** [comment]
Hi Wally, 
I have run some axisymmetric cases earlier for pressure distribution (Cp) computation over typical payload fairing configuration (with older versions of SU2). Pressure distribution seemed to be fine with earlier runs (and also it matched well with other codes). 
Issue popped up while trying to compute heat flux for axisymmetric problems. I have seen folks and students running their cases with axisymmetric formulation. Now the issue has surfaced clearly, it need to be fixed asap.
How is solution of the case, you have been trying? 


Best
Amit

**Comment by aeroamit** [comment]
Axisymmetric fixes in progress by Wally / Florian. Will repeat the test (axisymmetric) once fixes are done, I am closing it for now....

**Comment by WallyMaier** [comment]
@aeroamit I think the code should be performing. I think running your tests on the axisymmetric branch would helpful to get the effort pushed in :)

**Comment by aeroamit** [comment]
Sure @WallyMaier , I will get back with some results. I hope both NS and Turbulent parts are already attended / completed (including SST and SA model).......

**Comment by aeroamit** [comment]
I rolled back and had a look at the case again, it turned out to be issue with input given in one of the reference (I followed). I have changed the inputs from original reference and repeated the run (find the attached pdf [SU2_hflux_issue.pdf] https://github.com/su2code/SU2/files/5685357/SU2_hflux_issue.pdf)
) with finer grid. I also ran the case with alternate code. Pressure distribution match is good b/w the two but some differences are still there in heat flux distribution. The important thing is that I could not obtain a low dissipation 2nd order solution (perhaps this is common issue with such cases). I iterated to  higher E fix value and Venkat limiter coeff =0.01 to make it work (got some in between order of accuracy). Seems like there will be requirement of some knobs to tune the numerical dissipation for hypersonic cases to avoid carbuncle/instabilities.
I am closing it for now, as primary issue is sorted but perhaps new additions/improvements will be required.

Regards
Amit
